katacomb/katacomb/img_facade.py

```diff
@@ -78,17 +77,15 @@ def open_img(aips_path, mode=None, MF=False):
                            aips_path.aclass, aips_path.disk,
                            aips_path.seq, exists, err)
         except Exception:
-            raise (ValueError("Error calling newPAImage on '%s'" % aips_path),
-                   None, sys.exc_info()[2])
+            raise ValueError("Error calling newPAImage on '%s'" % aips_path)
```

Comment:
I'm not quite sure what this exception handling pattern in the old code was trying to achieve, but you've got a number of options in Python 3.

- As you've changed it, you'll see a tracebook first for the `Exception`, then it will say something like "During processing of the above exception, another exception occurred" and then show the traceback for the ValueError.
- If you catch with `except Exception as exc` then say `raise ValueError(...) from exc`, then it will be similar to the above, except the linking text will be something like "This was the direct cause of the following exception".
- If you want to completely hide the original `Exception` (which can make error messages clearer, but also make debugging harder if you needed to see exactly where the original was thrown) you can say `raise ValueError(...) from None`.

Reply:
I'm a lot happier with the python 3 way of doing this, and I think I'll leave it to how I've changed it. Previously the idea was to get an unknown exception from Obit and re-raise it locally with a known exception class, but with the traceback of the original exception for debugging.